### PR TITLE
Progress dot cleanup

### PR DESCRIPTION
**Pull request by Bjvanminnen**
This started out with me making a couple small changes to BubbleInterior, and then wanting to validate them. It ended with providing a bunch of storybooks for ProgressDot, which I think is actually super useful.

In regards to BubbleInterior, I realized that we don't need `courseOverviewPage` since `showLevelName` being true already implies `courseOverviewPage` being true. Also a lot of our falsey cases end up rendering the same (i.e. you can stick `undefined`, `''`, or `false` inside a span, and what you get is an empty span.

Some screenshots from storybook:

image

image

image

### CODE REVIEW

**Comment by Bjvanminnen** [comment]
One piece of feedback that could also be worthwhile, is if there are any major classes of ProgressDot that are missing from this list and could be added.

**Comment by mehalshah** [comment]
LGTM

File: apps/test/code-studio/ProgressDotTest.js [hunk 1]
```diff
@@ -60,14 +60,6 @@ describe('ProgressDot component tests', () => {
...
   });
 
-  it('bubble interior renders level title if not on overview page and not showing an icon', () => {
-    const result = ReactTestUtils.renderIntoDocument(
-      <BubbleInterior title={level.title} courseOverviewPage={false} showingIcon={false} showingLevelName={true} />
-    );
-
-    expect(ReactDOM.findDOMNode(result).innerHTML).to.equal('1');
-  });
-
```
Comment (Bjvanminnen) [inline apps/test/code-studio/ProgressDotTest.js]: Brent: Make sure you removed the right test, and also remove the `courseOverViewPage` prop from these tests, as it no longer is used.

**Comment by Bjvanminnen** [comment]
TODO: Add faded green dot (for status of passed instead of perfect)
Also add current level id (orange border)

**Pull request merged**